### DISCUSSION TITLE
Checking Business Rules in Csla 4.xx project

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3608] [Questions]
Our team has faced this scenario in the past.
We are asked to modify an existing project developed in Csla 4.xx.

We have a collection of objects and in one use case, the value of one property (in a child object) is changed which causes the associated business rules to execute, all good but we have a business rule that doesn't have a primary property (a per-type business rule) and we want this business rule to run every time the business (child)object is changed.
I understand we have to manually execute per-type business rules but where should we make this call? 

Kind Regards

**Comment by hurcane** [comment]
The per-type rule is on the child or the parent? I think some more specific detail about the rule conditions would help.

With collections, we frequently have to handle duplicate detection. This might be considered a candidate for an object-level rule because it is a duplicate object. However, we implement it using property-based rules in the child class and some code in the parent class to set a boolean property on the children that identifies whether or not they are a duplicate.

**Reply by Chicagoan2016** [reply]
Per-Type in the child object.
